tldr/cross_file_calls.py

```
@@ -3879,63 +4140,364 @@ def _build_php_call_graph(
+                        alias_map[last_segment] = full_module
+                elif imp.get('type') in ('import', 'use'):
+                    # Pre-build frozensets of function names for O(1) filter
+                    # lookups instead of O(N) any() scans at call-resolution time.
+                    #
+                    # NOTE: only/except lists store (name, arity) tuples from
+                    # the parser, but we match by name only because the call
+                    # extractor (_extract_elixir_file_calls) does not track
+                    # call-site argument counts. If arity tracking is added
+                    # to the call extractor, the frozensets below can be
+                    # replaced with arity-aware lookups.
+                    imp = dict(imp)  # shallow copy to avoid mutating shared data
```

Ruff (0.15.5)

[warning] `for` loop variable `imp` overwritten by assignment target

(PLW2901)